DevWs for Ecolink Intelligent Technology containing containing Ecolink DWZB1-ECO Contact Sensor

marchbold42 opened the pull request.

tpmanley committed:
[ICP-12506, ICP-12507] Zigbee White Color Temperature Bulb: Fix IKEA CT range (#20553)

When I recently added/updated these fingerprints I mistakenly used the wrong
UI Metadata vid so when these devices are joined the UI Metadata cannot be
found and so they default to a range of 0K to 100K.

https://smartthings.atlassian.net/browse/ICP-12506
https://smartthings.atlassian.net/browse/ICP-12507

msandstedt committed:
Revert "[CHAD-4341] Update generic Z-Wave Light bulb DTH for Z-Wave Light bulbs (#19205)"

This reverts commit 2c44449db7cea0e6750dc835a8146844a0c56912.

marchbold42 committed:
Initial commit of DWZB1-ECO device handler

marchbold42 committed:
- update DWZB1-ECO DH for testing

PKacprowiczS committed:
[WWST-6517] Fingerprint for Fibaro Walli Outlet (#20967)

* Added fingerprint for Fibaro Walli Outlet

* Tabs.

greens committed:
Revert "ICP-11631 Lock does not respond to battery polls after battery replac… (#17851)" (#20969)

This reverts commit 3b9aac6ed16a149a7919f2643d38e86f371c3c2a.

marchbold42 committed:
Release version

marchbold42 committed:
DevWs for Ecolink Intelligent Technology containing containing Ecolink DWZB1-ECO Contact Sensor

marchbold42 committed:
DevWs for Ecolink Intelligent Technology containing containing Ecolink DWZB1-ECO Contact Sensor

marchbold42 committed:
Added DWZB1-ECO to SmartSense Open/Closed Sensor. Added temp cluster to original DWZB1-ECO device handler.

PKacprowiczS commented on devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy:
```diff
@@ -0,0 +1,243 @@
+        attribute "tamper", "enum", ["detected", "clear"]
+
+		//fingerprints
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000,0001,0003,0020,0500,0B05", outClusters: "0019", manufacturer: "Ecolink", model: "DWZB1-ECO"//, deviceJoinName: "Ecolink Door/Window"
```
@marchbold42 I think there's no need for whole new DTH. Could you add fingerprint of your device to `smartsense-open-closed-sensor.groovy`? It should cover its whole functionality.

varzac replied:
Using an existing DTH is certainly preferable if possible.  It looks like this has some special tamper handling on the ias zone message.  But there is already ecolink related configuring in the other one and it looks like there should still be some ability to reuse that for this device as well.

PKacprowiczS commented on devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy:
```diff
@@ -0,0 +1,243 @@
+    return refreshCmds
+}
+
+def configure() {
```
`smartsense-open-closed-sensor.groovy` has already almost the same configuration as you've implemented. If binding with Poll Cluster is crucial for this model, you can define command for that more easily by utilizing `zigbee.addBinding(<add here Poll Cluster number>)`

PKacprowiczS reviewed (commented):
I'll ping @tpmanley @varzac for quick review, but I think they gonna agree with my recomendation.

PKacprowiczS commented:
@marchbold42 to continue with this request, please provide changes I've asked for on this PR;
- remove `ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy` file
- add your fingerprint to already existing DTH `smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy`
- define `deviceJoinName: "Ecolink Open/Closed Sensor"` in fingerprint
- if necessery, provide changes which are obligatory for device to work correctly in that DTH

Commit:
removed ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy file
added fingerprint to already existing DTH smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy
defined deviceJoinName: "Ecolink Open/Closed Sensor" in fingerprint

marchbold42 commented:
@PKacprowiczS I have performed the changes you request above. Please let me know if there is anything else that I need to do to close this issue. Thank you!

PKacprowiczS commented:
@marchbold42 if so, you haven't pushed it to your branch

Commit:
removed ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy file
added fingerprint to already existing DTH smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy for DWZB1-ECO sensor
defined deviceJoinName: "Ecolink Open/Closed Sensor" in fingerprint

Commit:
removed ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy file
added fingerprint to already existing DTH smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy for DWZB1-ECO sensor
defined deviceJoinName: "Ecolink Open/Closed Sensor" in fingerprint

marchbold42 commented:
@PKacprowiczS the changes have now been pushed.

shrakuma commented:
@greens @PKacprowiczS there is a conflict as i see it . what should be done to resolve the conflict
Thank you @marchbold42  for the chages pushed.

greens commented:
@marchbold42 you'll need to rebase your branch.

`git rebase --onto upstream/master`

assuming you've already set the upstream repo.

Also we're moving to more generic join names, which are the changes you'll be pulling in. We'd ask that you follow the new naming conventions which are [brand name] [device type].

Commit:
Merge branch 'devws-for-ecolink-intelligent-te_1' of https://github.com/marchbold42/SmartThingsPublic into devws-for-ecolink-intelligent-te_1

marchbold42 commented:
@greens @PKacprowiczS @shrakuma I did the rebase and pushed the new commit but it seems that there are still conflicts. What is the best course of action forward for this?

greens commented:
it looks like you've now just created a new DTH instead of adding a new fingerprint?

Perhaps it would be better to just start over.

marchbold42 commented:
Steven,

What steps should I take to start over?

Thank you,

Mike “Archy” Archbold
Systems Engineer
Ecolink Intelligent Technology, Inc. |  2055 Corte Del Nogal  Carlsbad, CA  92011
PH:  1-855 432-6546 ext. 112 | FAX: 1-760-431-8975
www.DiscoverEcolink.com<http://www.discoverecolink.com/>
[cid:image003.jpg@01D2D95D.DF90F250]
This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the Sender and delete all copies of this email. Please note that any views or opinions presented in this email are solely those of the author and do not necessarily represent those of Ecolink. The recipient should check this email and any attachments for the presence of viruses. Ecolink accepts no liability for any damage caused by any virus transmitted by this email.

From: Steven Green <notifications@github.com>
Sent: Monday, April 27, 2020 2:41 PM
To: SmartThingsCommunity/SmartThingsPublic <SmartThingsPublic@noreply.github.com>
Cc: Michael Archbold <archbold@discoverecolink.com>; Mention <mention@noreply.github.com>
Subject: Re: [SmartThingsCommunity/SmartThingsPublic] DevWs for Ecolink Intelligent Technology containing containing Ecolink DWZB1-ECO Contact Sensor (#21063)


it looks like you've now just created a new DTH instead of adding a new fingerprint?

Perhaps it would be better to just start over.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/21063#issuecomment-620250343>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOQTRJWMTNCIR63QOSQ752LROX3VTANCNFSM4KTL4GPQ>.

greens commented:
@marchbold42 do a fresh clone of the repository and submit a new PR where you just add your fingerprint to smartsense open/closed sensor

PKacprowiczS commented:
@marchbold42 let us know if you need any assitance here, it has been a while

marchbold42 commented:
Hello,

I have committed the fingerprint change into the repository on April 22 at https://github.com/marchbold42/SmartThingsPublic/blob/master/devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy

This is commit e93c9d51cd8ce3422f46d2660e9e6e93984ff4a5 under parent 856169f.

Please let me know if this change was made in the correct repository branch or if there is something I am missing here. Please let me know where to go from here.

Thank you,

Mike “Archy” Archbold
Systems Engineer
Ecolink Intelligent Technology, Inc. |  2055 Corte Del Nogal  Carlsbad, CA  92011
PH:  1-855 432-6546 ext. 112 | FAX: 1-760-431-8975
www.DiscoverEcolink.com<http://www.discoverecolink.com/>
[cid:image001.jpg@01D659CC.37B09C50]
This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the Sender and delete all copies of this email. Please note that any views or opinions presented in this email are solely those of the author and do not necessarily represent those of Ecolink. The recipient should check this email and any attachments for the presence of viruses. Ecolink accepts no liability for any damage caused by any virus transmitted by this email.

From: PKacprowiczS <notifications@github.com>
Sent: Wednesday, June 10, 2020 4:45 AM
To: SmartThingsCommunity/SmartThingsPublic <SmartThingsPublic@noreply.github.com>
Cc: Michael Archbold <archbold@discoverecolink.com>; Mention <mention@noreply.github.com>
Subject: Re: [SmartThingsCommunity/SmartThingsPublic] DevWs for Ecolink Intelligent Technology containing containing Ecolink DWZB1-ECO Contact Sensor (#21063)


@marchbold42<https://github.com/marchbold42> let us know if you need any assitance here, it has been a while

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/21063#issuecomment-641948392>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOQTRJW7FBQELAXROUCTERLRV5W4LANCNFSM4KTL4GPQ>.

greens commented:
@marchbold42 so you'll want to make a pull request from your fork to our repo. You can do that by going to https://github.com/SmartThingsCommunity/SmartThingsPublic , selecting the "pull requests" tab, clicking "new pull request", and then "compare across forks". Then specify your fork and branch.

marchbold42 commented:
> @marchbold42 so you'll want to make a pull request from your fork to our repo. You can do that by going to https://github.com/SmartThingsCommunity/SmartThingsPublic , selecting the "pull requests" tab, clicking "new pull request", and then "compare across forks". Then specify your fork and branch.

@greens Got it! Thank you for the instructions. I have submitted the new pull request .

greens commented:
@marchbold42 Can you tag me on that PR so I can find it?

marchbold42 committed:
Replaced spaces with tabs for Ecolink DWZB1-ECO finger print

greens commented:
@marchbold42 I think you want to `git rm ecolink-zigbee-door-window-sensor.groovy` and commit that, otherwise that file will continue to show up in this PR.

You'll also want to `git rebase upstream/master` and resolve those conflicts.

marchbold42 commented:
@greens I completed the rebase and removed ecolink-zigbee-door-window-sensor.groovy as far as I can tell. Please let me know if there is anything else I need to do.

greens commented:
@marchbold42 on this PR or a different one?

marchbold42 commented:
@greens I did not make a new pull request as when I go to the pull request screen on github it will only take me here and will not allow me to create a new pull request. What I did was clone the repository on my desktop, did the rebase, made sure the ecolink device handler was not present, and pushed it back up to github.

greens commented:
@marchbold42 so your local directory and git's reckoning of what files should be in it are different. You still need to call `git rm` on the file to have git forget about it even if you've already deleted it in your OS. You can see in the Files tab above that there are still 2 affected files in this PR.

marchbold42 committed:
Delete ecolink-zigbee-door-window-sensor.groovy

marchbold42 commented:
@greens Alright I think I have it this time. I appreciate the patience and guidance on this.

greens reviewed (changes requested):
It looks good in the diff, but the merge dialog is still telling me there are conflicts.

It _might_ work to `git fetch upstream master` and then `rebase` again.

marchbold42 commented:
@greens I attempted the fetch and rebase on my local repository and pushed it back to gitub afterwards. Is the conflict resolved?

greens commented:
@marchbold42 not sure what the issue is at this point, I've just opened a PR with your new fingerprint here and I'll make sure it gets in: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/38312

marchbold42 commented:
@greens Thank you for all of your help!

greens commented:
@marchbold42 I've merged my PR and the fingerprints should be live on production on Tuesday next week.

marchbold42 commented:
@greens Thank you so much!

KKlimczukS commented:
@greens
Shouldn't this PR be closed since #38312 was merged?
cc: @marchbold42
